modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -3016,13 +3062,19 @@ private void scheduleTransactionRowAsyncCleanup(UUID txId, RowId rowId, Transact
 
         // Both normal cleanup and single row cleanup are using txsPendingRowIds map to store write intents.
         // So we don't need a separate method to handle single row case.
-        txManager.executeCleanupAsync(() ->
-                inBusyLock(busyLock, () -> storageUpdateHandler.handleTransactionCleanup(txId, txState == COMMITED, commitTimestamp))
-        ).exceptionally(e -> {
-            LOG.warn("Failed to complete transaction cleanup command [txId=" + txId + ']', e);
+        CompletableFuture<?> future = rowCleanupMap.compute(rowId, (k, v) -> {
+            CompletableFuture<Void> rowCleanup = txManager.executeCleanupAsync(() ->
+                    inBusyLock(busyLock, () -> storageUpdateHandler.handleTransactionCleanup(txId, txState == COMMITED, commitTimestamp))
+            ).exceptionally(e -> {
+                LOG.warn("Failed to complete transaction cleanup command [txId=" + txId + ']', e);
```

Comment:
Why do we expect handleTransactionCleanup to fail? In other words why it's warn and not error?
I believe that it shouldn't be possible to proceed with given tx if we failed to cleanup the writeIntent, thus we should rollback current transaction.